2020/02/21/consolidate-multiple-sftp-accounts-on-one-master-account.md

```diff
@@ -0,0 +1,154 @@
+
+
+* Create a user group for SFTP accounts
+
+```bash
+$ addgroup sftpusers
+```
+
+* Configure the following settings in sshd_config (this enables an SFTP account and defaults the location as the home path)
+
+```bash
+$ vi /etc/ssh/sshd_config
+
+Subsystem       sftp    internal-sftp
+
+Match Group sftpusers
+    ChrootDirectory /home/%u
+    AllowTCPForwarding no
+    X11Forwarding no
+    ForceCommand internal-sftp
+```
+
+* Restart SSH Server to apply changes
+
+```bash
+$ /etc/init.d/ssh restart
+```
+
+* Create an SFTP user account for a facility and place in a folder on the home path to receive data
+
+```bash
+useradd $USERNAME
+usermod -g sftpusers -s /usr/sbin/nologin $USERNAME
+mkdir -p /home/$USERNAME
+mkdir -p /home/$USERNAME/INPUT_PATH/
+chown -R root:root /home/$USERNAME
+```
+
+As the number of facilities increases, additional communication points are added. If the permissible amount of points is exceeded on the app by even one, it will not continue its service until more blocks are paid for. The implementation of this process will directly benefit our client.
+
+### Multiple Accounts Mounts to One Account
+
+The goal here is to point the data from many facilities to one location. Yet, using a single account and path for multiple sites’ data will result in a breach in security and privacy. However, mounting the receiving path of a facility’s data onto a single master account to then “mount point” with a unique facility name takes care of this issue. The process continues on to consolidate files from individual paths on a master account in one place where the application picks up messages for further processing.
+
+
+Both the SFTP accounts as well as master account should be attached to one group. This will permit individual accounts to write on the master account-mounted path. In turn, the master account can read files from the same location. This location now has administrative rights of both the SFTP user as well as shared group permission. Group permission of the mounted folder is set to sftpgroup and user permission is set to facility account.
+
+
+A script is written to automate the list of actions in order to create an SFTP account, mount it at the master account path, and add fstab entries to save the mount in the case of a reboot. The script not only saved much time, but also avoided human error when creating accounts for all facilities in addition to future usage.
+
+* Create master account, sftpgroup and mount paths
+
+```bash
+
+# create master user account
+$ adduser master
+$ passwd master
+
+# Add master user to sftpgroup group
+usermod -a -G sftpgroup master
+getent group | grep master
+
+# MOUNT_PATH and sub folders
+mkdir -p /home/master/MOUNT_PATH/{Input,Pickup,Backup,Archive}
+chown -R master:master /home/master/MOUNT_PATH
+```
+
+* Automated script to create sftp account and mount folders
+
+```bash
+#!/bin/bash
+
+# sh create_sftp_account.sh sftpfacilityone FACILITY_ONE
+
+USERNAME=$1
```

Comment:
Here I see you set the `USERNAME` variable, but I still think that's a bad idea since at least my local shell has `USERNAME` set. Why not use a lowercase variable that won't be confusing?

Reply:
Agreed, using lower case is best practice for custom variables.
sftpuser=$1
facility_name=$2